aten/src/ATen/native/mkldnn/OpContext.cpp

```diff
@@ -7,6 +7,69 @@ namespace at {
 namespace native {
 namespace mkldnn {
 
+#define ATTR_FUNC(NAME)                              \
+  [](std::vector<c10::optional<at::Scalar>> scalars, \
+     c10::optional<std::string> algorithm) {         \
+    return ideep::attr_t::fuse_##NAME();             \
+  }
+
+AttrFunction attr_func_leaky_relu =
+    [](std::vector<c10::optional<at::Scalar>> scalars,
+       c10::optional<std::string> algorithm) {
+      TORCH_CHECK(scalars.size() == 1 && scalars[0].has_value(), "leaky_relu is expected to have one scalar input: negative_slope");
+      auto alpha_value = scalars[0].value().to<float>();
+      return ideep::attr_t::fuse_relu(1.0, alpha_value);
+    };
+
+AttrFunction attr_func_hardtanh =
+    [](std::vector<c10::optional<at::Scalar>> scalars,
+       c10::optional<std::string> algorithm) {
+      TORCH_CHECK(
+          scalars.size() == 2 &&
+              std::all_of(
+                  scalars.begin(),
+                  scalars.end(),
+                  [](c10::optional<at::Scalar> item) {
+                    return item.has_value();
+                  }),
+          "hardtanh is expected to have two scalar input: min_val and max_val");
+
+      auto lower_bound_value = scalars[0].value().to<float>();
+      auto upper_bound_value = scalars[1].value().to<float>();
+      return ideep::attr_t::fuse_clamp(lower_bound_value, upper_bound_value);
+    };
+
+AttrFunction attr_func_gelu = [](std::vector<c10::optional<at::Scalar>> scalars,
+                                 c10::optional<std::string> algorithm) {
+  TORCH_CHECK(
+      algorithm.has_value(),
+      "gelu is expected to have one str input: algorithm");
+  dnnl::algorithm gelu_type;
+  if (algorithm.value() == "none") {
+    gelu_type = dnnl::algorithm::eltwise_gelu_erf;
+  } else if (algorithm.value() == "tanh") {
+    gelu_type = dnnl::algorithm::eltwise_gelu_tanh;
+  } else {
+    TORCH_CHECK(
+        false, "Unsupported gelu algorithm: ", algorithm.value());
```

Comment:
https://pytorch.org/docs/stable/generated/torch.nn.GELU.html

nn.GELu only supports "none" and "tanh". So I'd recommend using `TORCH_INTERNAL_ASSERT`

Reply:
Converted to `TORCH_INTERNAL_ASSERT`